djproxy/headers.py

```diff
@@ -3,14 +3,17 @@
 
 
 class HeaderDict(dict):
-
     """A dict containing header, value pairings."""
 
     @staticmethod
     def _normalize_django_header_name(header):
         """Unmunge header names modified by Django."""
-        # Remove HTTP_ prefix.
-        new_header = header.rpartition('HTTP_')[2]
+        prefix = 'HTTP_'
+        # Remove HTTP_ prefix if present.
+        if header.startswith(prefix):
```

Comment:
Great catch.

Let's make this a little more confident and enforce a default operating context so we don't have to deal with that else clause:

```python
new_header = header

# Standard headers are prefixed with HTTP, which needs to be removed
# to get the true header name.
if new_header.startswith('HTTP_'):
   # I prefer rpartition here still, but I'll leave it to you.
    _, _, new_header = header.rpartition('HTTP_')
```

Let's also update the comment to explain why not what we're doing.

Reply:
We would have to use `partition()` instead of `rpartition()` to get the right result.  Once we know that the header starts with `"HTTP_"` it seems more clear to me to just remove the prefix directly.

This is the counterexample for why `rpartition()` doesn't work:
```
>>> "HTTP_X_HTTP_METHOD".partition("HTTP_")
('', 'HTTP_', 'X_HTTP_METHOD')
>>> "HTTP_X_HTTP_METHOD".rpartition("HTTP_")
('HTTP_X_', 'HTTP_', 'METHOD')
```
For the special headers like `"CONTENT_TYPE"`, this shows why we can't use `partition()` without checking if the `"HTTP_"` prefix is present:
```
>>> "CONTENT_TYPE".partition("HTTP_")
('CONTENT_TYPE', '', '')
>>> "CONTENT_TYPE".rpartition("HTTP_")
('', '', 'CONTENT_TYPE')
```